vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java

```diff
@@ -333,15 +393,40 @@ private void processGamePiece(GamePiece piece, Command command) {
       // Place new piece on the map where the old piece was
       // Then position it at the same position in the stack the old piece was
 
+
+      // Place the new Piece.
+
+      // Here we use a workaround for placing pieces into decks. If 2 decks are defined
+      // with the same x,y position, the piece placement will not be able to determine
+      // the target deck. So 1st move the deck of the piece being replaced to -1, -1
+      // then put it back to its original position
+
+      final Stack stack = piece.getParent();
+      final Deck deck = (Deck) stack;
```

Comment:
Isn't this going to throw a `ClassCastException` if `stack` is not a `Deck`? You do an `instanceof` check just below this, so either that's possible or the check is redundant.